tests/setuptab.test.js

```diff
@@ -580,4 +587,260 @@ describe('Form drag-and-drop', () => {
+      },
+    };
+    fetchValidation.mockResolvedValue(
+      [[Object.keys(spec.args), 'invalid because']]
+    );
+
+    const { findByLabelText, findByTestId } = renderSetupFromSpec(spec);
+    const setupForm = await findByTestId('setup-form');
+    const setupInput = await findByLabelText(RegExp(`${spec.args.arg1.name}`));
+
+    const fileDragEvent = createEvent.dragEnter(setupInput);
+    // `dataTransfer.files` normally returns a `FileList` object. Since we are
+    // defining our own dataTransfer.files we are also creating an object
+    // with properties that mimic FileList object
+    const fileValue = {};
+    Object.defineProperties(fileValue, {
+      path: { value: 'foo.txt' },
+      length: { value: 1 },
+    });
+    Object.defineProperty(fileDragEvent, 'dataTransfer', {
+      value: { files: [fileValue] }
+    });
+    fireEvent(setupInput, fileDragEvent);
+
+    expect(setupForm).not.toHaveClass("dragging");
+    expect(setupInput).toHaveClass("input-dragging");
+
+    const fileDropEvent = createEvent.drop(setupInput);
+    Object.defineProperty(fileDropEvent, 'dataTransfer', {
+      value: { files: [fileValue] }
+    });
+    fireEvent(setupInput, fileDropEvent);
+
+    expect(setupInput).not.toHaveClass("input-dragging");
+    expect(setupForm).not.toHaveClass("dragging");
+    expect(setupInput).toHaveValue( 'foo.txt' );
+  });
+
+  test('Drag enter/leave of a file sets .input-dragging class on input', async () => {
+    const spec = {
+      module: `natcap.invest.${MODULE}`,
+      args: {
+        arg1: {
+          name: 'Workspace',
+          type: 'directory',
+        },
+        arg2: {
+          name: 'AOI',
+          type: 'vector',
+        },
+      },
+    };
+    fetchValidation.mockResolvedValue(
+      [[Object.keys(spec.args), 'invalid because']]
+    );
+
+    const { findByLabelText, findByTestId } = renderSetupFromSpec(spec);
+    const setupInput = await findByLabelText(RegExp(`${spec.args.arg1.name}`));
+
+    const fileDragEnterEvent = createEvent.dragEnter(setupInput);
+    // `dataTransfer.files` normally returns a `FileList` object. Since we are
+    // defining our own dataTransfer.files we are also creating an object
+    // with properties that mimic FileList object
+    const fileValue = {};
+    Object.defineProperties(fileValue, {
+      path: { value: 'foo.txt' },
+      length: { value: 1 },
+    });
+    Object.defineProperty(fileDragEnterEvent, 'dataTransfer', {
+      value: { files: [fileValue] }
+    });
+    fireEvent(setupInput, fileDragEnterEvent);
+
+    expect(setupInput).toHaveClass("input-dragging");
+
+    const fileDragLeaveEvent = createEvent.dragLeave(setupInput);
+    Object.defineProperty(fileDragLeaveEvent, 'dataTransfer', {
+      value: { files: [fileValue] }
+    });
+    fireEvent(setupInput, fileDragLeaveEvent);
+
+    expect(setupInput).not.toHaveClass("input-dragging");
+  });
+
+  test('Drag and drop on a disabled input element.', async () => {

```

Comment:
I thought the intent here would be to do the drop and expect no updated value in the input, but this test does a "drag & leave" with no drop.

Reply:
Good catch! Fixed here: https://github.com/natcap/invest-workbench/pull/92/commits/744114de22a529559e2d1273d1697074bbf05b30